File: content/snippets/cpp/language_basics_file_paths_and_directories.cpp
#include <filesystem>
#include <iostream>

int main() {
    namespace fs = std::filesystem;

    fs::path base = "demo_folder";
    fs::path file_path = base / "sub" / "data.txt";

    fs::create_directories(file_path.parent_path());

    std::cout << "file path: " << file_path.string() << "\n";
    std::cout << "parent path: " << file_path.parent_path().string() << "\n";
    std::cout << "directory exists: " << fs::exists(file_path.parent_path()) << "\n";
    return 0;
}

File: content/snippets/typescript/language_basics_file_paths_and_directories.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";
import { mkdirSync, existsSync } from "fs";
import path from "path";

function main(): void {
  const base = "demo_folder";
  const filePath = path.join(base, "sub", "data.txt");
  const parent = path.dirname(filePath);

  mkdirSync(parent, { recursive: true });

  console.log("file path:", filePath);
  console.log("parent path:", parent);
  console.log("directory exists:", existsSync(parent));
}

main();
setImmediate(() => {});
